src/confighttp.cpp

```diff
@@ -464,71 +587,66 @@ namespace confighttp {
       fileTree.add_child("apps", sorted_apps);
 
       pt::write_json(config::stream.file_apps, fileTree);
+      proc::refresh(config::stream.file_apps);
+
+      outputTree.put("status", true);
+      send_response(response, outputTree);
     }
     catch (std::exception &e) {
       BOOST_LOG(warning) << "SaveApp: "sv << e.what();
-
-      outputTree.put("status", "false");
-      outputTree.put("error", "Invalid Input JSON");
-      return;
+      bad_request(response, request, e.what());
     }
-
-    outputTree.put("status", "true");
-    proc::refresh(config::stream.file_apps);
   }
 
   /**
    * @brief Delete an application.
    * @param response The HTTP response object.
    * @param request The HTTP request object.
+   *
+   * @api_examples{/api/apps/9999| DELETE| null}
    */
   void
   deleteApp(resp_https_t response, req_https_t request) {
     if (!authenticate(response, request)) return;
 
     print_req(request);
 
-    pt::ptree outputTree;
-    auto g = util::fail_guard([&]() {
-      std::ostringstream data;
+    int index;
 
-      pt::write_json(data, outputTree);
-      response->write(data.str());
-    });
-    pt::ptree fileTree;
+    pt::ptree outputTree;
     try {
+      pt::ptree fileTree;
+      pt::ptree newApps;
       pt::read_json(config::stream.file_apps, fileTree);
       auto &apps_node = fileTree.get_child("apps"s);
-      int index = stoi(request->path_match[1]);
+      index = stoi(request->path_match[1]);
 
-      if (index < 0) {
-        outputTree.put("status", "false");
-        outputTree.put("error", "Invalid Index");
+      if (std::clamp(index, 0, static_cast<int>(apps_node.size()) - 1) != index) {
```

Comment:
```suggestion
      if (index < 0 && index >= static_cast<int>(apps_node.size())) {
```

Congrats, c++ standard sends love!
"If `lo` is greater than `hi`, the behavior is undefined."

This will be the case when `size()` returns `0`.